tasks/create-product-sbom/create-product-sbom.yaml

```diff
@@ -20,19 +20,35 @@ spec:
       description: Workspace to save the product-level SBOM to.
   results:
     - name: productSBOMPath
-      description: Relative path to the created product-level SBOM in the data workspace.
+      description: >-
+        Relative path to the directory containing the created product-level SBOM
+        in the data workspace.
   steps:
     - name: create-sbom
-      image: quay.io/konflux-ci/release-service-utils:8684920ccae6c73bd9f3f23367490a9c04653a09
+      image: quay.io/konflux-ci/release-service-utils:221d71a4f6b1a50b36b685aa20d86d7df9de33fc
       script: |
         #!/usr/bin/env bash
         set -eux
 
-        SBOM_FILE="product_sbom.json"
-        SBOM_PATH="$(dirname "$(params.dataJsonPath)")/${SBOM_FILE}"
-        OUTPUT_PATH=$(workspaces.data.path)/${SBOM_PATH}
-
+        tmp_sbom="$(mktemp)"
         create_product_sbom --data-path "$(workspaces.data.path)/$(params.dataJsonPath)" \
-          --output-path "$OUTPUT_PATH"
+          --output-path "$tmp_sbom"
+
+        product_name="$(jq -r '.packages[0].name' "$tmp_sbom")"
+        product_version="$(jq -r '.packages[0].versionInfo' "$tmp_sbom")"
+
+        # Replace whitespace with dashes
+        normalized_name="$(echo -n "${product_name}" | tr '[:space:]' '-')"
+
+        sbom_dir="product-sboms"
+        # the combination of name + version is later used as an ID in Atlas
+        sbom_path="${sbom_dir}/${normalized_name}-${product_version}.json"
+
+        # takes into account the subdirectory of the data.json if any
+        subdir_sbom_path="$(dirname "$(params.dataJsonPath)")/${sbom_path}"
+
+        output_path=$(workspaces.data.path)/${subdir_sbom_path}
+        mkdir -p "$(dirname "$output_path")"
+        cp "$tmp_sbom" "$output_path"
```

Comment:
This is done this way because the filename in the output path depends on the contents of the SBOM (the name and version of the product) and I wanted to avoid the complexity of duplicating the data extraction from the data.json. I added a comment for clarity and used `mv` to make it more obvious.

Reply:
got it, thanks!